main.js

```diff
@@ -125,22 +137,22 @@ function addClicks() {
 
 }
 
-function getCoords(addr) {
+async function getCoords(addr) {
 
     let urlAddress = encodeURIComponent(addr);
-   
-    const coordObj = {};  
 
-    fetch(`https://nominatim.openstreetmap.org/search/?q=${urlAddress}&format=json`)
+    const coordObj = {};
+
+    let coords = await fetch(`https://nominatim.openstreetmap.org/search/?q=${urlAddress}&format=json`)
         .then(data => data.json())
         .then(addrObj => {
-            
             coordObj.lat = parseFloat(addrObj[0].lat)
             coordObj.lng = parseFloat(addrObj[0].lon)
+            console.log(coordObj);
+            return {lat: coordObj.lat, lon: coordObj.lng};
```

Comment:
here, we return the lat and lng from the coordObj and return it as an object with the properties lat and lon (could be lng to keep things consistent in your code, although it may be a little confusing after having lng in so many places). So this will return this object and the coords variable will turn into:
```js
coords = {
  lat: coordObj.lat,
  lon: coordObj.lng
}
```